### PR TITLE
Make the inserter in the site editor behave as a popover

### PR DESCRIPTION
**Pull request by youknowriad**
This matches the behavior of the inserter in the post editor and fixes some a11y issues for the inserter (constrained tabbing, closes when the focus is outside...)

### CODE REVIEW

File: packages/edit-site/src/components/editor/popover-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/**
```
Comment (youknowriad) [inline packages/edit-site/src/components/editor/popover-wrapper.js:1]: This file is duplicated between edit-post and edit-site. Separately, I'd like to factorize that and transform it to a React hook in `@wordpress/compose` if possible.

**Comment by jasmussen** [comment]
I haven't had a chance to follow too deeply with the discussions here, so I only vaguely recall a conversation about this behavior on another ticket, which of course I can't find. I CC'ed a couple of people in case they have context for whether this was intentional or not.

File: packages/edit-site/src/components/editor/popover-wrapper.js (hunk 1 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/editor/popover-wrapper.js:1]: That sounds a good idea 👍

Comment (ntsekouras) [reply packages/edit-site/src/components/editor/popover-wrapper.js:1]: Actually it's used in `edit-widgets` as well, so it's triplicate.

**Review by ntsekouras — APPROVED** [review]
Tested and works well! Depending on the priority of this to land you can try to make the hook to be reused or in a follow up. 👍

**Pull request merged**